modules/nw-cudn-cr.adoc

```diff
@@ -0,0 +1,146 @@
+<6> For `Layer2` topology types the following specifies config details for the `subnet` field:
++
+* The subnets field is optional.
+* The subnets field is of type `string` and accepts standard CIDR formats for both IPv4 and IPv6.
+* The subnets field accepts one or two items. For two items, they must be of a different family. For example, subnets values of `10.100.0.0/16` and `2001:db8::/64`.
+* `Layer2` subnets can be omitted. If omitted, users must configure IP addresses for the pods. As a consequence, port security only prevents MAC spoofing.
+* The `Layer2` `subnets` field is mandatory when the `ipamLifecycle` field is specified.
++
+.. Create a YAML file, such as `cluster-layer-three-udn.yaml`, to define your request for a `Layer3` topology as in the following example:
++
+[source, yaml]
+----
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+metadata:
+  name: <cudn_name> # <1>
+spec:
+  namespaceSelector:
+    matchExpressions:
+    - key: kubernetes.io/metadata.name <2>
+      operator: In <3>
+      values: ["<my_value>"] <4>
+  network:
+    topology: Layer3 # <5>
+    layer3: # <6>
+      role: Primary # <7>
+      subnets: # <8>
+        - cidr: 10.100.0.0/16
+----
+<1> Name of your `UserDefinedNetwork` resource. This should not be `default` or duplicate any global namespaces created by the Cluster Network Operator (CNO).
+<2> Specifies the label key to match.
+<3> Specifies the operator. Valid values include: `In`, `NotIn`, `Exists`, and `DoesNotExist`.
+<4> Specifies an array of string values. If the `operator` value is either `Exists` or `DoesNotExist`, the value array must be empty.
+<5> The `topology` field describes the network configuration; accepted values are `Layer2` and `Layer3`. Specifying a `Layer3` topology type creates a layer 2 segment per node, each with a different subnet. Layer 3 routing is used to interconnect node subnets.
+<6> This field specifies the topology configuration. Valid values are `layer2` or `layer3`.
+<7> Specifies a `Primary` or `Secondary` role. `Primary` is the only `role` specification supported in {product-version}.
+<8> For `Layer3` topology types the following specifies config details for the `subnet` field:
++
+* The `subnets` field is mandatory.
+* The type for the `subnets` field is `cidr` and `hostSubnet`:
+** `cidr` is the cluster subnet and accepts a string value.
+** `hostSubnet` specifies the nodes subnet prefix that the cluster subnet is split to.
+** For IPv6, only a `/64` length is supported for `hostSubnet`.
++
+. Apply your request by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f <my_layer_two_udn.yaml>
+----
++
+Where `<my_layer_two_udn.yaml>` is the name of your `Layer2` or `Layer3` configuration file.
+
+. Verify that your request is successful by running the following command:
++
+[source, terminal]
+----
+$ oc get clusteruserdefinednetwork <cudn_name> -o yaml
```

Comment:
We had some bugs opened, where `oc` is used and the CRD manifest has bad fields names, for example: 
- `mt` instead of `mtu` [[1]](https://issues.redhat.com/browse/OCPBUGS-38706)
- `hostsubnet` instead of `hostSubnet` [[2]](https://issues.redhat.com/browse/OCPBUGS-38901)
It result in unexpected results due to the fact `oc` does not perform schema validation by default [3] - it silently removing fields with bad names and create the object.

Should we specify in the --validate=true in this example? 
Is so, I suppose we should update `UserDefiendNetwork` examples as well.

[1] https://issues.redhat.com/browse/OCPBUGS-38706
[2] https://issues.redhat.com/browse/OCPBUGS-38901
[3] From OC help:
```
 --validate='ignore':
        Must be one of: strict (or true), warn, ignore (or false).              "true" or "strict" will use a schema to validate
        the input and fail the request if invalid. It will perform server side validation if ServerSideFieldValidation
        is enabled on the api-server, but will fall back to less reliable client-side validation if not.                "warn" will
        warn about unknown or duplicate fields without blocking the request if server-side field validation is enabled
        on the API server, and behave as "ignore" otherwise.            "false" or "ignore" will not perform any schema
        validation, silently dropping any unknown or duplicate fields.
```
kubectl set `validation=true` by default.

Reply:
Added --validate=true to this command.